content/openpubkey-and-sigstore.md

```diff
@@ -0,0 +1,223 @@
+
+When someone wants to verify the signature:
+
+1. They verify that the code signing certificate was signed correctly by Fulcio
+1. They verify the identity and issuer of the certificate match the identity and issuer they expect.
+1. They verify the signature against the artifact
+1. At this point, they know that the identity in the certificate signed the artifact, as vouched for by Fulcio (who checked the identity from the identity provider)
+
+Note that the “keyless” flow in sigstore is optional, many other flows exist and are also supported, and these other flows may be better in some cases!
+The keyless flow is most comparable to that of OpenPubKey, so that’s what I’m explaining here.
+
+This whole process sounds complex, and it is!
+The user experience is very simple though, the entire process can be wrapped up into two commands:
+
+```
+$ cosign sign
+$ cosign verify
+```
+
+Also note that this is still a simplified view.
+There are a few other things going on here with transparency logs that you don’t need to understand quite yet.
+
+# How OpenPubKey Works
+
+In OpenPubKey, the user goes through a similar, albeit simpler flow:
+
+1. A user logs into their identity provider (Google, Github, Microsoft, etc.).
+1. The user requests an “identity token” from this provider, that they can use to assert their identity to a third party.
+*DIFFERENCE ALERT* As part of this step, the user embeds some information into the ‘nonce’ field of the request. This is an implementation detail of 1. the OIDC protocol. (more on this below)
+1. The identity provider hands back the signed token, including the signed nonce, which contains that special information.
+1. The user then uses this token directly as a certificate - the nonce field contains information about the user’s signing key, so it binds their identity to that key so they can sign artifacts right away!
+1. The user attaches this token alongside the signature, similar to a certificate.
+
+For verification:
+
+1. They verify that the identity token was signed by the identity provider (Google, GitHub, etc.)
+1. They verify the signature against the artifact
+1. At this point, they know that the identity in the certificate signed the artifact, as vouched for by the identity provider
+
+This explanation is a bit over-simplified.
+OpenPubKey actually uses two tokens (the original ID token which contains the nonce, and their own PK token which effectively wraps the ID token and contains the public key.
+The full setup looks like this:
+
+![](/images/openpubkey.png)
+
+OpenPubKey also contains support for expiration - the default in this mode is for these certificates to be valid for two weeks.
+The paper doesn’t really specify what happens after that, or how to handle revocation, but we can imagine ways these can be constructed on top of this protocol.
+
+This effectively eliminates the Fulcio component by using the token directly as a certificate.
```

Comment:
I'd rephrase this as "this effectively removes the OIDC part of Fulcio (but not its transparency log)" or so